✨(keycloak) add ralph client for generating access tokens

wilbrdt opened the pull request:
Add a `ralph` client to `fun-mooc` realm.

The `ralph` client has been configured with the specific audience `http://localhost:8100` but it can be changed through the Keycloak interface.
Two users have been created for this client:
- `ralph_admin:funfunfun` with the scope `all`
- `ralph_learner:moocmooc` with the scope `statements/read/mine`

To get the access token, you can use the following command:
```
curl -X POST -d "grant_type=password" -d "client_id=ralph" -d "client_secret=bcef3562-730d-4575-9e39-63e185f99bca" -d "username=ralph_admin" -d "password=funfunfun" http://localhost:8080/auth/realms/fun-mooc/protocol/openid-connect/token

wilbrdt committed:
✨(keycloak) add ralph client for generating access tokens

Add a `ralph` client to `fun-mooc` realm, with the creation of two users.

quitterie-lcs commented on bin/realm:
```diff
@@ -37,7 +43,41 @@ kcadm.sh create "clients/${client_uuid}/roles" -r "${realm}" -s name=editor
+kcadm.sh create users -r "${realm}" -s username="${ralph_learner_user}" -s email="${ralph_learner_email}" -s enabled=true
+kcadm.sh set-password -r "${realm}" --username "${ralph_learner_user}" --new-password "${ralph_learner_password}"
+# Add role for ralph user
+kcadm.sh add-roles -r "${realm}" --uusername "${ralph_learner_user}" --cclientid "ralph" --rolename "statements/read/mine"
```
```suggestion
kcadm.sh add-roles -r "${realm}" --uusername "${ralph_learner_user}" --cclientid "ralph" --rolename "statements/read/mine"

```

quitterie-lcs commented on data/keycloak/clients/ralph.json:
```diff
@@ -0,0 +1,35 @@
+      }
+    ]
+  }
+  
```
```suggestion

```

quitterie-lcs commented on env.d/keycloak:
```diff
@@ -14,10 +14,19 @@ DB_PASSWORD=password
+RALPH_ADMIN_TEST_USER_EMAIL=ralph_admin@example.org
+RALPH_LEARNER_TEST_USER_NAME=ralph_learner
+RALPH_LEARNER_TEST_USER_PASSWORD=moocmooc
+RALPH_LEARNER_TEST_USER_EMAIL=ralph_learner@example.org
```
```suggestion
RALPH_LEARNER_TEST_USER_EMAIL=ralph_learner@example.org

```

quitterie-lcs reviewed (commented):
SGTM! 
<img src="https://media2.giphy.com/media/l2JJrEx9aRsjNruhi/giphy.gif"/>

wilbrdt committed:
🔨(project) add network creation makefile rule

Bootstrapping the project should create non existing external networks.
Adding a rule for that.

SergioSim reviewed (commented):
Nice!
Have only one question regarding the role-to-scope mapping.
![image](https://media.giphy.com/media/xT5LMqTuWw95jo72Cs/giphy.gif)

wilbrdt commented on .circleci/config.yml:
```diff
@@ -85,6 +85,9 @@ jobs:
       - run:
           name: Create external potsie network
           command: docker network create potsie
+      - run:
+          name: Create external ralph network
+          command: docker network create ralph
```
Indeed! Good catch, thanks!

wilbrdt commented on env.d/keycloak:
```diff
@@ -14,10 +14,19 @@ DB_PASSWORD=password
 GRAFANA_CLIENT_SECRET=fa9e98ee-61a1-4092-8dac-1597da0c1bb0
+
+# Ralph
+RALPH_ADMIN_TEST_USER_NAME=ralph_admin
```
Good improvement, thanks!
